chain/transaction_test.go

```
@@ -496,3 +503,41 @@ func equalTxData(r *require.Assertions, expected chain.TransactionData, actual c
 	}
 	r.Equal(expected.UnsignedBytes(), actual.UnsignedBytes(), msgAndArgs...)
 }
+
+// updateReferenceBlockData regenerates the reference hex and JSON files with the current implementation
+// Only runs when UPDATE_TEST_DATA=1 is set eg: UPDATE_TEST_DATA=1 go test ./chain/...
+func updateReferenceTxData() {
```

Creating separate functions for this duplicates the code we are comparing the test vector against.

Would it make sense to write to file directly in the tests comparing against the hardcoded types ie. if running a normal test, perform the `require.Equal(...)` check, if this environment variable is specified, then skip the require check and overwrite the file.

I think this reduce the code duplication and make it a bit easier to read ie. overwrite this test vector we're comparing against this way of re-generating it here.

Separately, it seems potentially error prone to use this environment variable to overwrite the test vectors during the test run.

This means that if the environment variable is set incorrectly, then our tests would pass, overwrite the test vector, and unless we have a diff check after running the tests, we may not notice.

What do you think of separating the command to update the test vectors from the tests themselves?

---

I've been weighing between generating test data using commands or scripts because that's generally how this project automates things. I decided to have this function so that everything would be in one file, to avoid the 'internalized' knowledge that you need to run a specific script or command and which command. I relied on review and diff to avoid silent errors.

After this, I'm not sure what value we get with this change - it seems to me we're just adding complexity. I'd revert this change.